pepys_admin/merge.py

```diff
@@ -0,0 +1,743 @@
+                        .filter(master_table.name == slave_entry.name)
+                        .all()
+                    )
+                    n_name_results = len(search_by_name_results)
+
+                    if n_name_results == 0:
+                        ids_added.append({"id": guid, "name": slave_entry.name})
+                        make_transient(slave_entry)
+                        master_store.session.add(slave_entry)
+                    elif n_name_results == 1:
+                        ids_modified.append(
+                            {
+                                "from": guid,
+                                "to": getattr(search_by_name_results[0], primary_key),
+                                "name": slave_entry.name,
+                                # Data can never be changed here, because there's only one field (name) and that's what we search by
+                                "data_changed": False,
+                            }
+                        )
+                        setattr(
+                            slave_entry,
+                            primary_key,
+                            getattr(search_by_name_results[0], primary_key),
+                        )
+                        slave_store.session.add(slave_entry)
+                        slave_store.session.commit()
+                    else:
+                        assert (
+                            False
+                        ), "Fatal assertion error: multiple entries in master reference table with same name"
+                elif n_results == 1:
+                    ids_already_there.append({"id": guid, "name": slave_entry.name})
+                else:
+                    assert (
+                        False
+                    ), "Fatal assertion error: multiple entries in master reference table with same GUID"
+
+    return {
+        "already_there": ids_already_there,
+        "added": ids_added,
+        "modified": ids_modified,
+    }
+
+
+def merge_all_metadata_tables(master_store, slave_store, merge_change_id):
+    """Merge *most* metadata tables from the slave_store into the master_store, using the merge_change_id
+    as the change_id for any modifications occuring as part of the merge.
+
+    Note: this does not merge the Datafile, Synonym, Log or Change tables - these are handled separately.
+    """
+    master_store.setup_table_type_mapping()
+    metadata_table_objects = master_store.meta_classes[TableTypes.METADATA]
+
+    metadata_table_names = [obj.__name__ for obj in metadata_table_objects]
+
+    # Crude ordering system for now (TODO: Improve)
+    # Put Platform first, then Sensor, then rest of them
+    metadata_table_names.remove("Platform")
+    metadata_table_names.remove("Sensor")
+    metadata_table_names.insert(0, "Sensor")
+    metadata_table_names.insert(0, "Platform")
+
+    # Remove various entries for now - deal with those separately later
+    metadata_table_names.remove("Datafile")
```

Comment:
I'll have a think about this. I _think_ you're probably right, but I need to revisit why I did it that way in the first place.

Reply:
I looked into this, and because the `id_results` variable is no longer returned from functions like `merge_all_metadata_tables` (I think you changed this when you moved to a class system, and I agree with it), we need to do the datafiles separately as we need the `id_results['added']` list to pass to the measurement merging function.